src/migrations/index.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+import Promise from 'bluebird';
+import AsyncStorage from '@react-native-community/async-storage';
+import _ from 'lodash/fp';
+import type {Migrations} from './types';
+
+const applyMigration = (predicate, migration) => async (asyncStorage: typeof AsyncStorage) => {
+  const allASKeys = await asyncStorage.getAllKeys();
+
+  return Promise.mapSeries(allASKeys, async key => {
+    if (predicate(key)) {
+      const value = await asyncStorage.getItem(key);
+      const nextValue = await migration(value, key);
+      if (value !== nextValue) {
+        await asyncStorage.setItem(key, nextValue);
+      }
+    }
+  });
+};
+
+const applyMigrations = (asyncStorageVersionKey, migrations) => async (
+  asyncStorage: typeof AsyncStorage
+) => {
+  const currentASVersion = (await asyncStorage.getItem(asyncStorageVersionKey)) || 1;
+  return _.pipe(
+    _.toPairs,
+    _.map(_.update('0', _.parseInt(10))),
+    _.filter(([version]) => version > currentASVersion),
+    _.sortBy('0'),
+    vm =>
+      Promise.mapSeries(vm, async ([version, [predicate, migration]]) => {
+        await applyMigration(predicate, migration)(asyncStorage);
+        await asyncStorage.setItem(asyncStorageVersionKey, currentASVersion + 1);
```

Comment:
```suggestion
        await asyncStorage.setItem(asyncStorageVersionKey, version);
```
C'est pas forcement currentASVersion + 1. Surtout si tu passes de la v1 à la v3, `currentASVersion + 1 === 2`  et non `3`.